datalad/support/annexrepo.py

```diff
@@ -12,30 +12,31 @@
 
 """
 
-from os import linesep
-from os.path import join as opj, exists, relpath, islink, realpath, lexists
-import logging
 import json
-import re
+import logging
 import os
+import re
 import shlex
+from os import linesep
+from os.path import join as opj, exists, islink, realpath, lexists
 from subprocess import Popen, PIPE
+
 #import pexpect
 
 from functools import wraps
 
 from six import string_types
 from six.moves import filter
 from six.moves.configparser import NoOptionError
-from six.moves.urllib.parse import quote as urlquote
 
 from ..dochelpers import exc_str
 from ..utils import auto_repr
-from .gitrepo import GitRepo, normalize_path, normalize_paths, GitCommandError
+from datalad.support.gitrepo import GitRepo, normalize_path, normalize_paths, GitCommandError
```

Comment:
yes, let's discuss, agree and do uniformly may be in a single or few dedicated PRs...  
I am not sure though that I would really prefer absolute imports within e.g. tests deep under, when they nearly always just need to import stuff from upstairs (regardless how deep it is).  Makes it easy to identify what functionality from the actual submodule they are testing.  And for that reason PEP8 even was adjusted to reflect such use cases:  while generally recommending absolute imports it suggests that "explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose".  In my case I would hate to `from datalad.crawler.nodes.misc import X` within tests under there whenever I could just do `from ..misc import X`... anyways -- it would be a point of discussion -- feel welcome to open a new issue for that, meanwhile let's keep it locally consistent.